src/components/ResultsPanel/ResultsPanel.js

```diff
@@ -16,43 +16,64 @@ export default class ResultsPanel extends React.Component {
   constructor(props){
     super(props);
     this.scrollTo = () => {};
+    this.state = { needBackToTop: false };
+  }
+
+  handleScroll = (event) => {
+    this.setState({ needBackToTop: event.target.scrollTop > 500 });
   }
 
   getLocals() {
     const {
-      results,
-      searchedValue,
-      loadingResults
-    } = this.props;
+      props: {
+        results,
+        searchedValue,
+        loadingResults
+      },
+      state: {
+        needBackToTop
+      }
+    } = this;
     return {
       results,
       searchedValue,
-      loadingResults
+      loadingResults,
+      needBackToTop,
+      handleScroll: this.handleScroll
     };
   }
 
-  template({ results, searchedValue, loadingResults }){
+  template({ results, searchedValue, loadingResults, handleScroll, needBackToTop }){
     return (
         <Panel
           className="results-panel"
           type='floating'
-          header={{ title: (searchedValue ? 'Results for : ' + searchedValue : 'Welcome')}}
+          header={{ title: (searchedValue ? 'Results for : ' + searchedValue : 'Welcome') }}
         >
           <ScrollView
             easing='easeInOutQuad'
             scrollX={false}
             scrollPropagation={false}
-            style={{ position: 'absolute', width: '100%', maxHeight:'100%'}}
+            style={{ position: 'absolute', width: '100%', maxHeight:'100%' }}
+            onScroll={handleScroll}
           >
           {(scrollTo) => {
             this.scrollTo = scrollTo;
             return (
-                <List list={results} loading={loadingResults} />
-              );
-              }
+              <List list={results} loading={loadingResults} />
+            );
+            }
           }
           </ScrollView>
-          <button className="back-to-top" style={{position:'absolute'}} onClick={()=> this.scrollTo(0, 0, Math.min(1500, results.length * 100))}><i className="fa fa-arrow-up"></i></button>
+          {needBackToTop && (
+            <button
+              className="back-to-top"
+              style={{ position:'absolute' }}
+              onClick={()=> this.scrollTo(0, 0, results ? Math.min(1500, results.length * 100) : 1500)}
```

Comment:
as per convention, you should move this to `getLocals`:
```js
// ...
scrollDuration: results ? Math.min(1500, results.length * 100) : 1500
// ...
```

no logic in templates FTW!